RFC-0010-access-control-filters-masks.md

```diff
@@ -0,0 +1,205 @@
+    {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Bulk method for getting column masks for a subset of columns in a table.
+     * <p>
+     * Each mask must be a scalar SQL expression of a type coercible to the type of the column being masked. The expression
+     * must be written in terms of columns in the table.
+     *
+     * @return a mapping from columns to masks, or an empty map if not applicable. The keys of the return Map are a subset of {@code columns}.
+     */
+    default Map<ColumnMetadata, ViewExpression> getColumnMasks(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName, List<ColumnMetadata> columns)
+    {
+        return Collections.emptyMap();
+    }
+```
+
+SystemAccessControl.java
+```java
+
+    /**
+     * Get row filters associated with the given table and identity.
+     * <p>
+     * Each filter must be a scalar SQL expression of boolean type over the columns in the table.
+     *
+     * @return a list of filters, or empty list if not applicable
+     */
+    default List<ViewExpression> getRowFilters(Identity identity, AccessControlContext context, CatalogSchemaTableName tableName)
+    {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Bulk method for getting column masks for a subset of columns in a table.
+     * <p>
+     * Each mask must be a scalar SQL expression of a type coercible to the type of the column being masked. The expression
+     * must be written in terms of columns in the table.
+     *
+     * @return a mapping from columns to masks, or an empty map if not applicable. The keys of the return Map are a subset of {@code columns}.
+     */
+    default Map<ColumnMetadata, ViewExpression> getColumnMasks(Identity identity, AccessControlContext context, CatalogSchemaTableName tableName, List<ColumnMetadata> columns)
+    {
+        return Collections.emptyMap();
+    }
+```
+
+ViewExpression class to hold a filter/mask expression
+
+```java
+public ViewExpression(String identity, Optional<String> catalog, Optional<String> schema, String expression)
+```
+
+Analysis.java will hold filters and masks for the table with additional methods
+
+```java
+void registerTableForRowFiltering(QualifiedObjectName table, String identity)
+
+boolean hasRowFilter(QualifiedObjectName table, String identity)
+
+void addRowFilter(Table table, Expression filter)
+
+List<Expression> getRowFilters(Table node)
+
+void registerTableForColumnMasking(QualifiedObjectName table, String column, String identity)
+
+boolean hasColumnMask(QualifiedObjectName table, String column, String identity)
+
+void addColumnMask(Table table, String column, Expression mask)
+
+Map<String, Expression> getColumnMasks(Table table)
+```
+
+#### Additional information
+
+1. What modules are involved
```

Comment:
I think you're missing double whitespace at line endings, so this is not rendering correctly - 
![image](https://github.com/user-attachments/assets/2ed8ba51-7ed9-4b7f-b469-73958171ac94)